### DISCUSSION TITLE
Tasmota with a couple of network interfaces (ethernet and wifi) command changes

### DISCUSSION THREAD

**Discussion by urechi** [arendst/Tasmota#14168] [Ideas]
Status5 (NET) for ESP32 with ethernet and wifi should return data for two networks

**Comment by urechi** [comment]
For `Status5` command, this is what we get now:
```
{
   "StatusNET":{
      "Hostname":"**B9**D2**40-4928-eth",
      "IPAddress":"192.168.1.35",
      "Gateway":"192.168.1.1",
      "Subnetmask":"255.255.255.0",
      "DNSServer1":"192.168.1.1",
      "DNSServer2":"0.0.0.0",
      "Mac":"**:B9:**:D2:**:43",
      "Webserver":2,
      "HTTP_API":1,
      "WifiConfig":4,
      "WifiPower":17.0
   }
}
```

And this is what would be nice to get in the future:
```
{
   "StatusNET":{
       "eth": {
          "Hostname":"**B9**D2**40-4928-eth",
          "IPAddress":"192.168.1.35",
          "Gateway":"192.168.1.1",
          "Subnetmask":"255.255.255.0",
          "DNSServer1":"192.168.1.1",
          "DNSServer2":"0.0.0.0",
          "Mac":"**:B9:**:D2:**:43"
      },
      "wifi": {
          "Hostname":"**B9**D2**42-4928-wifi",
          "IPAddress":"192.168.1.36",
          "Gateway":"192.168.1.1",
          "Subnetmask":"255.255.255.0",
          "DNSServer1":"192.168.1.1",
          "DNSServer2":"0.0.0.0",
          "Mac":"**:B9:**:D2:**:42"
      },
      "Webserver":2,
      "HTTP_API":1,
      "WifiConfig":4,
      "WifiPower":17.0,
      "ExternalIPAddress: "1.1.1.1"
   }
}
```

Please note the ExternalIPAddress too.

A huge thank you to everyone contributing to this project, you are all rock stars!

**Comment by Jason2866** [comment]
What is/should be `"ExternalIPAddress: "1.1.1.1"` ?

**Comment by urechi** [comment]
@Jason2866 that should be the [public IP address](https://en.wikipedia.org/wiki/IP_address#Public_address). Useful if there is a remote location.

```
curl https://ipecho.net/plain ; echo
```
or
```
wget -qO- https://ipecho.net/plain ; echo
```
Hope that helps.

**Reply by sfromis** [reply]
It is really, really not recommended to expose Tasmota devices to the wild internet, the "security" is just not up to that. I make sure that my devices definitely do not have a public IP address ;-)

**Reply by barbudor** [reply]
That also means depending on a specific external service. If it goes down, those versions of Tasmota will have problems.

You can do the same with a rule and WebQuery. No need to have this built in Tasmota.
Really not wanted.

**Reply by urechi** [reply]
> It is really, really not recommended to expose Tasmota devices to the wild internet, the "security" is just not up to that. I make sure that my devices definitely do not have a public IP address ;-)

I agree that exposing tasmota devices to the internet is not a good idea, I work in the cyber security space. My use case is around tasmota devices talking to an exposed MQTTS broker.

**Comment by s-hadinger** [comment]
I don't think we will change Status 5 because it would break existing automation. 

However you should easily be able to add this feature with some Berry scripting. Especially since it requires an https request on the internet to get the router's IP

**Reply by urechi** [reply]
@s-hadinger I get it for the external IP address but does having both wifi and eth network details returned by Status 5 make sense, I guess?

**Reply by barbudor** [reply]
What about the following :
- If only 1 interface is present, no change on the message
- If 2 are present, existing syntax is used for Wifi but Ethernet use a child "eth" section (or same fields with ".eth" appended)

This would not break any existing implementation

**Comment by urechi** [comment]
Also when you have two active network connections at the same time, `IPAddress<x>` would set networking data for which network? eth or wifi? I can test that, I've got a few ZB-GW03 bridges.

**Reply by barbudor** [reply]
It is generally a bad idea to have 2 interfaces on the same subnetwork. When you can't configured routing tables, the result is very OS dependant. I have seen different weird behavior on different OS such as :
- ingoing and out going packets not going though the same interface
- positive ping results on the B interface when it's actually down but the A is up
I'm not talking about strange and exotic OS but some variants of Linux and Windows.

Also with 2 active network interfaces, you shouldn't have 2 default gateway. That shouldn't be allowed as it also leads to bad behavior,

**Reply by urechi** [reply]
In regards to the breaking changes here we could do:
`IPAddress1 192.168.1.100` set device IP address for `wifi`
`IPAddress1 192.168.1.100 wifi` set device IP address for `wifi`
`IPAddress1 192.168.1.100 eth` set device IP address for `eth`

**Reply by urechi** [reply]
@barbudor I am currently playing with a `ZB-GW03` which got an `ESP32-D0WDQ6-V3` chip. The web server is up for both wifi and eth and couldn't spot any issues so far. That being said both are going through the same gateway.

**Reply by urechi** [reply]
Currently I've got
`WIFI` with DHCP -> `192.168.1.32`
`ETH` with DHCP -> `192.168.1.35`

Then, I run this command:
```
backlog IPAddress1 192.168.1.200; IPAddress2 192.168.1.1; IPAddress3 255.255.255.0; IPAddress4 192.168.1.1; restart 1
```
After restart, the `WIFI` network interface got `192.168.1.200` and the web server is also available at `192.168.1.35` on the ethernet interface.

**Reply by urechi** [reply]
```
14:37:55.422 CMD: Status 5
14:37:55.423 SRC: WebConsole from 192.168.1.141
14:37:55.425 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '5'
14:37:55.428 RSL: STATUS5 = {"StatusNET":{"Hostname":"*********-4928-eth","IPAddress":"192.168.1.35","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"**:**:**:**:**:**","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
```

It looks like the ethernet interface is the preferred network which is totally normal.

**Reply by urechi** [reply]
Next, `backlog IPAddress1 0.0.0.0; restart 1` and a new personal hotspot on my iPhone. The WIFI is now connected to the personal hotspot SSID. The iPhone has a small DHCP address pool that can give out 16 addresses (172.20.10.0-172.20.10.15). Ethernet is still connected to my local 192.168.1.0/24 network.
At this point, the two network interfaces are connected to separate/different networks.

```
14:57:37.939 CMD: Status 5
14:57:37.941 SRC: WebConsole from 192.168.1.141
14:57:37.942 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '5'
14:57:37.945 RSL: STATUS5 = {"StatusNET":{"Hostname":"*********-4928-eth","IPAddress":"192.168.1.35","Gateway":"172.20.10.1","Subnetmask":"255.255.255.240","DNSServer1":"172.20.10.1","DNSServer2":"0.0.0.0","Mac":"*********","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
```

Please note from the `Status 5` response, the IPAddress is the ethernet IP and then Gateway, Subnetmask and DNSServer1 are populated from the WIFI interface. This is a bug and might confirm @barbudor assumptions above. Of course, unless a proper packet capture is done we can't confirm the bad behavior...

**Reply by urechi** [reply]
Next, unplug the ethernet cable and restart. Doing a few pings from my iPhone in order to find the IP address assigned by the personal hotspot DHCP and I can access the web UI from my phone browser at http://172.20.10.6.

```
15:22:12.408 CMD: Status 5
15:22:12.410 SRC: WebConsole from 172.20.10.1
15:22:12.411 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '5'
15:22:12.414 RSL: STATUS5 = {"StatusNET":{"Hostname":"*******-4928","IPAddress":"172.20.10.6","Gateway":"172.20.10.1","Subnetmask":"255.255.255.240","DNSServer1":"172.20.10.1","DNSServer2":"0.0.0.0","Mac":"*********","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
```
Everything looks good at this point.

Now plugging the ethernet cable back and without a restart:
```
15:31:52.108 CMD: Status 5
15:31:52.110 SRC: WebConsole from 172.20.10.1
15:31:52.111 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '5'
15:31:52.114 RSL: STATUS5 = {"StatusNET":{"Hostname":"*********-4928-eth","IPAddress":"192.168.1.35","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"************","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
```
Looks good.

Next `restart 1`
```
15:37:45.104 CMD: Status 5
15:37:45.105 SRC: WebConsole from 172.20.10.1
15:37:45.107 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '5'
15:37:45.110 RSL: STATUS5 = {"StatusNET":{"Hostname":"*********-4928-eth","IPAddress":"192.168.1.35","Gateway":"172.20.10.1","Subnetmask":"255.255.255.240","DNSServer1":"172.20.10.1","DNSServer2":"0.0.0.0","Mac":"*******","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
```
And the bug is back.

**Comment by urechi** [comment]
Equivalent to `Wifi` command should we have `Ethernet` = 0 disable ; 1 enable?

**Reply by urechi** [reply]
Ignore this, apologies I just saw that's already implemented [here](https://tasmota.github.io/docs/Commands/#ethernet).

**Comment by Jason2866** [comment]
It is NO problem to have two network adapters in one device in the same network with the same gateway, as long there is no traffic routed between the adapters (which is the case here).

**Reply by barbudor** [reply]
@urechi Can you please elaborate your use case ?
- Ethernet only or Simultaneous Ethernet+Wifi
- What would be the network configuration ? Both interface in same subnet ? Or separated subnet ?
- In the 2nd case, please alaborate why you would like to have the same device to belong to 2 different networks ?

**Comment by budulinek** [comment]
Hi everybody. I am another user of ZB-GW03 . To start with, let me thank everybody involved in the Tasmota project! Great piece of software!

OK, so I can also confirm that web server works on both eth and wifi IPs. This is my sonfiguration, both IPs are from DHCP:
<img width="372" alt="tasmota" src="https://user-images.githubusercontent.com/6001151/148472794-2a2ceaaa-e74c-4271-a10f-dc9289a56242.png">

This is my `Status 5` , same result when connected through eth interface on `http://10.10.10.226/cs?` and wifi interface on `http://10.10.10.223/cs?` 

```
{
   "StatusNET":{
      "Hostname":"tasmota-D2FC6C-7276-eth",
      "IPAddress":"10.10.10.226",
      "Gateway":"10.10.10.1",
      "Subnetmask":"255.255.255.0",
      "DNSServer1":"10.10.10.1",
      "DNSServer2":"0.0.0.0",
      "Mac":"XXX",
      "Webserver":2,
      "HTTP_API":1,
      "WifiConfig":4,
      "WifiPower":17.0
   }
}
```

and `IPAddress1` results also same for both interfaces. Even if I am connected through wifi (on `http://10.10.10.223/cs?` ), I got eth IP in the output (in the brackets):
```
{
   "IPAddress1":"0.0.0.0 (10.10.10.226)"
}
```

Now let's set static IP through `IPAddress1 10.10.10.220`  and then `restart 1`....

After boot:
<img width="349" alt="tasmota2" src="https://user-images.githubusercontent.com/6001151/148478246-d51876c2-41a3-439b-ae36-4f29405a780d.png">

and `Status 5` 

```
{
   "StatusNET":{
      "Hostname":"tasmota-D2FC6C-7276-eth",
      "IPAddress":"10.10.10.226",
      "Gateway":"10.10.10.1",
      "Subnetmask":"255.255.255.0",
      "DNSServer1":"10.10.10.1",
      "DNSServer2":"0.0.0.0",
      "Mac":"XXX",
      "Webserver":2,
      "HTTP_API":1,
      "WifiConfig":4,
      "WifiPower":17.0
   }
}
```
and `IPAddress1` 
```
{
   "IPAddress1":"10.10.10.220 (10.10.10.226)"
}
```
To summarize, this is the current behaviour of the these commands:

- command `Status 5` always shows in "IPAddress" field: **ethernet IP**
- command `IPAddress1` always shows: **static wifi IP** (0.0.0.0 if not set) + **DHCP ethernet IP** in the brackets
- command `IPAddress1 X.X.X.X` always sets: **wifi IP**

This is really confusing. `Status 5` "prefers" ethernet and shows eth IP even if I am connected to Tasmota via wifi interface. On the other side, `IPAddress1 X.X.X.X`  prefers wifi. Even if I am connected via ethernet (and with `Wifi 0` disabled), it only sets wifi IP. Impossible to set static eth IP. 

My suggestion: `Status 5` and `IPAddress1` should show / set IP address of the interface to which I am connected. There is no "preferred" interface.

If connected through eth interface:

- command `Status 5` shows in existing syntax: **ethernet** (IP, hostname, mac...)
- command `IPAddress1` shows: **static ethernet IP** (0.0.0.0 if not set) + **DHCP ethernet IP** in the brackets
- command `IPAddress1 X.X.X.X` sets: **ethernet IP**

If connected through wifi interface:

- command `Status 5` shows in existing syntax: **wifi** (IP, hostname, mac...)
- command `IPAddress1` shows: **static wifi IP** (0.0.0.0 if not set) + **DHCP wifi IP** in the brackets
- command `IPAddress1 X.X.X.X` sets: **wifi IP**


`Status 5` the existing syntax will always show info for the interface to which we are connected. If we really want to see info for the other interface, add new section.
`IPAddress1 X.X.X.X` should set the IP for the interface in use. If we really want to change eth IP from within wifi interface (and vice versa), add new commands.

**Comment by barbudor** [comment]
Dear all
Can you please elaborate your use case ?
- Ethernet only or Simultaneous Ethernet+Wifi
- What would be the network configuration ? Both interface in same subnet ? Or separated subnet ?
- In the 2nd case, please alaborate why you would like to have the same device to belong to 2 different networks ?

**Reply by budulinek** [reply]
Hi,
My use case:

- Most of the time, I will use Ethernet only (wifi disabled through `Wifi 0` ). But occasionaly also Ethernet + Wifi.
- Same subnet. I see no reason for separate subnets.

**Reply by budulinek** [reply]
Also, we need to take into account that `Wifi 1` + `Ethernet 1` is a default setup. So we need to deal with both "Ethernet + Wifi" and "Ethernet only" use cases.

**Comment by urechi** [comment]
Hi @barbudor, first thanks for looking into this.

I've got two internet connections (ISPs), one supplied by Virgin Media and another by Three (wifi 3G modem).
Virgin Media is used for the ethernet and Three for the wifi connection to my Tasmota ZB Hub. If the main power goes down, there is a UPS that will power my ZB HUB along with the wifi 3G modem. So on power outages, my Zigbee network (except the Zigbee routers) will still be able to communicate with my Home Automation hub (through MQTT) which is deployed in the cloud.

- Simultaneous Ethernet + Wifi, with ethernet the default interface unless there is a power outage.
- Separated subnets
- For high availability, redundancy and security

**Reply by antoniomorsello** [reply]
Hi all I need to use the tasmota firmware only ethernet, when I execute the "wifi 0" command after restart 1 it always gives me the ethernet ip address but I can no longer connect

**Comment by Jason2866** [comment]
Wifi and Ethernet have different IP addresses. You have to connect to the Ethernet address.

**Comment by antoniomorsello** [comment]
Thanks for the answer, I tried but I can't connect to the ip address anymore, I have to reset everything and flash the firmware again

**Reply by sfromis** [reply]
For me it just works to use Ethernet with Wifi off. I even have a Berry script running to automatically disable Wifi after boot, if Ethernet is available. And to enable Wifi again if I disconnect Ethernet and reboot.

Without that, I have both different IP addresses, and both different host names responding at the same time, but Tasmota favoring Wifi over Ethernet for outbound connections.

**Reply by antoniomorsello** [reply]
I would like it to work ethernet only

**Reply by sfromis** [reply]
To enable/disable: https://tasmota.github.io/docs/Commands/#wifi

**Reply by antoniomorsello** [reply]
do you recommend Wifi manager 0 ?

**Reply by sfromis** [reply]
No. I was linking the command `Wifi`, not `WifiConfig`.

**Reply by antoniomorsello** [reply]
I've already tried Wifi 0 and it doesn't work, I lose the programming

**Comment by antoniomorsello** [comment]
I've already tried Wifi 0 and it doesn't work, I lose the programming

**Reply by antoniomorsello** [reply]
I tried now and when I put Wifi 0 in the console and restart from ipscan I find the ip address of my tasmota LAN but when I write the ip address on the browser I get Err_connectio_refused

**Reply by phoenix83** [reply]
I found the same: with wifi disabled (0) the webui isn't accessible (via Ethernet). I'm wondering if the web server isn't started until after a successful wifi connection.
Did you ever find the solution?

**Reply by s-hadinger** [reply]
Thus is a very old thread and the problem you mention is supposedly fixed. Please open a new issue

**Comment by nikito7** [comment]
Ota Upgrade fails over Ethernet.  Maybe related to safeboot

**Reply by sfromis** [reply]
I just tested an OTA upgrade with Wifi turned off, and of course Ethernet on. No failure. Better supply good details instead of merely saying "fails" and "maybe".

**Reply by nikito7** [reply]
ESP32 S3
W5500

It just break, no logs

I need to do serial flash again

**Reply by sfromis** [reply]
That was no useful info about what happens. And "just breaks" are again words which hides why that was your conclusion.

**Reply by nikito7** [reply]
[23:41:01]23:41:01.064 CMD: Wifi 0
[23:41:01]23:41:01.065 SRC: WebConsole from 10.2.0.70
[23:41:01]23:41:01.065 CMD: Grp 0, Cmd 'WIFI', Idx 1, Len 1, Pld 0, Data '0'
[23:41:01]23:41:01.067 RSL: RESULT = {"Wifi":"OFF","Mode":"HT20"}
[23:41:02]23:41:01.938 CFG: Saved, Count 30, Bytes 4096
[23:41:09]23:41:08.946 HTP: Tools
[23:41:10]23:41:09.696 HTP: Main Menu
[23:41:11]23:41:10.697 HTP: Firmware Upgrade
[23:41:11]23:41:11.448 HTP: Upgrade started
[23:41:11]23:41:11.448 SRC: WebGui from 10.2.0.70
[23:41:11]23:41:11.449 CMD: Grp 0, Cmd 'OTAURL', Idx 1, Len 56, Pld -99, Data 'http://ota.tasmota.com/tasmota32/release/tasmota32s3.bin'
[23:41:11]23:41:11.460 CFG: CR 387/699, Busy 0
[23:41:11]23:41:11.461 RSL: RESULT = {"OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32s3.bin"}
[23:41:12]23:41:11.514 SRC: WebGui from 0.0.0.0
[23:41:12]23:41:11.515 CMD: Grp 0, Cmd 'UPGRADE', Idx 1, Len 1, Pld 1, Data '1'
[23:41:12]23:41:11.516 RSL: RESULT = {"Upgrade":"Version 14.2.0 from http://ota.tasmota.com/tasmota32/release/tasmota32s3.bin"}
[23:41:12]23:41:11.836 CFG: Saved, Count 31, Bytes 4096
[23:41:14]23:41:13.600 APP: Restarting
[23:41:15]
[23:41:15]00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on USB port
[23:41:15]00:00:00.251 HDW: ESP32-S3 v0.1 (PSRAM)
[23:41:15]00:00:00.261 CFG: Loaded from NVS, Count 31
[23:41:15]00:00:00.261 CFG: CR 387/699, Busy 0
[23:41:15]00:00:00.263 SRC: Restart
[23:41:15]00:00:00.263 Project tasmota - Tasmota Version 14.2.0(release-safeboot)-3_0_4(2024-08-14T12:37:14)
[23:41:15]00:00:00.273 WARNING This version does not support persistent settings
[23:41:23]00:00:08.522 APP: Boot Count 14
[23:42:03]
Serial port closed!

**Reply by nikito7** [reply]
> That was no useful info about what happens. And "just breaks" are again words which hides why that was your conclusion.

This is new to me.

And just hangs.

If safeboot don't know spi gpios, it's normal...

Enabling wifi fixed it

Thank you anyway 

```
00:00:08.522 APP: Boot Count 14

wifi

00:00:35.312 CMD: wifi

00:00:35.312 SRC: Serial

00:00:35.312 CMD: Grp 0, Cmd 'WIFI', Idx 1, Len 0, Pld -99, Data ''

00:00:35.313 RSL: RESULT = {"Wifi":"OFF","Mode":"low rate"}

wifi 1
      00:00:49.817 CMD: wifi 1

00:00:49.817 SRC: Serial

00:00:49.817 CMD: Grp 0, Cmd 'WIFI', Idx 1, Len 1, Pld 1, Data '1'

00:00:49.818 RSL: RESULT = {"Wifi":"ON","Mode":"low rate"}

00:00:50.317 WIF: Checking connection...

00:00:50.317 WIF: Attempting connection...

00:00:51.000 WIF: Connecting to AP1 ssn-x Channel 1 BSSId ***** in mode HT20 as tasmota-DBDEDC-7900...

00:00:51.537 WIF: IPv4 10.1.0.64, mask 255.255.255.0, gateway 10.1.0.1

00:00:52.518 WIF: IPv6 Local fe80::2658:7cff:fedb:dedc%st1

00:00:52.714 WIF: Checking connection...

00:00:52.714 WIF: Connected

00:00:52.971 OTA: Propagating upload

00:00:52.972 HTP: Web server active on tasmota-DBDEDC-7900 with IP addres 10.1.0.64

00:00:53.518 WIF: IPv6 Global fd10:9:9:1:2658:7cff:fedb:dedc

00:00:53.966 RSL: INFO1 = {"Info1":{"Module":"ESP32S3","Version":"14.2.0release-safeboot)","FallbackTopic":"cmnd/DVES_DBDEDC_fb/","GroupTopic":"mnd/tasmotas/"}}

00:00:53.977 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"asmota-DBDEDC-7900","IPAddress":"10.1.0.64","IP6Global":"fd10:9:9:1:26587cff:fedb:dedc","IP6Local":"fe80::2658:7cff:fedb:dedc%st1"}}

00:00:53.989 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset,"BootCount":15}}

00:00:55.215 UPL: http://ota.tasmota.com/tasmota32/release/tasmota32s3.bn 14.2.0(release-safeboot)

00:00:55.841 OTA: Connected in 330 ms, stack low mark 4444

00:01:27.031 RSL: UPGRADE = {"Upgrade":"Successful. Restarting"}

00:01:28.281 APP: Restarting



00:00:00.251 CMD: Fall back to serial port, no SOF packet detected on US port

00:00:00.251 HDW: ESP32-S3 v0.1 (PSRAM)

00:00:00.253 UFS: FlashFS mounted with 304 kB free

00:00:00.266 CFG: Loaded from File, Count 31

00:00:00.273 QPC: Count 1

00:00:00.274 CFG: CR 387/699, Busy 0

00:00:00.275 SPI: Bus1 using GPIO37(CLK), GPIO48(MOSI) and GPIO33(MISO)

00:00:00.277 TFS: File 'mcp23x.dat' not found

00:00:00.288 MCP: Invalid template

00:00:00.288 TYA: Active=0

00:00:00.290 ROT: Mode 1

00:00:00.293 BRY: GC from 3692 to 3492 bytes, objects freed 4/20 (in 0 m) - slots from 36/61 to 28/61

00:00:00.308 CFG: No '*.autoconf' file found

00:00:00.310 BRY: GC from 5295 to 3764 bytes, objects freed 8/35 (in 0 m) - slots from 55/76 to 36/76

00:00:00.310 BRY: Berry initialized, RAM used 3764 bytes

00:00:00.323 BRY: No 'preinit.be'

00:00:00.324 SRC: Restart

00:00:00.325 Project tasmota - Tasmota Version 14.2.0(release-tasmota32)3_0_4(2024-08-14T12:40:10)

00:00:00.360 ETH: Attempting connection...

00:00:00.365 BRY: No 'autoexec.be'

00:00:02.087 ETH: Connected at 100Mbps Full Duplex, Mac 26:58:7C:DB:DE:D, Hostname tasmota-DBDEDC-7900-eth

00:00:02.108 ETH: IPv4 10.1.0.31, mask 255.255.255.0, gateway 10.1.0.1

00:00:02.487 NTP: Sync time...

00:00:02.527 WIF: DNS resolved '2.pool.ntp.org' (194.117.47.44) in 39 ms

00:00:02.550 RTC: UTC 2024-08-18T22:55:45Z, DST 2024-03-31T02:00:00, STD2024-10-27T03:00:00

23:55:45.001 RTC: Synced by NTP

23:55:45.058 HTP: Web server active on tasmota-DBDEDC-7900-eth with IP adress 10.1.0.31

23:55:45.607 RSL: INFO1 = {"Info1":{"Module":"ESP32S3","Version":"14.2.0release-tasmota32)","FallbackTopic":"cmnd/DVES_DBDEDC_fb/","GroupTopic":cmnd/tasmotas/"}}

23:55:45.618 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{Hostname":"tasmota-DBDEDC-7900-eth","IPAddress":"10.1.0.31"}}}

23:55:45.630 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU""BootCount":14}}

23:55:45.631 TFS: File 'autoexec.bat' not found

23:55:46.063 ETH: IPv6 Local fe80::2458:7cff:fedb:dedc

23:55:48.511 QPC: Reset

23:55:49.063 ETH: IPv6 Global fd10:9:9:1:2458:7cff:fedb:dedc

23:55:50.505 APP: Boot Count 14

23:55:50.507 BRY: GC from 5062 to 4076 bytes, objects freed 15/45 (in 1 s) - slots from 76/91 to 50/91

23:55:50.508 RSL: STATE = {"Time":"2024-08-18T23:55:50","Uptime":"0T00:0:09","UptimeSec":9,"Heap":268,"SleepMode":"Dynamic","Sleep":50,"LoadAvg"12,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45}}

23:55:51.220 CFG: Saved, Count 32, Bytes 4096

23:55:55.510 RSL: config = {"ip":"10.1.0.31","dn":"Tasmota","fn":["Tasmoa",null,null,null,null,null,null,null],"hn":"tasmota-DBDEDC-7900-eth","mc":"24587CDBDEDC","md":"ESP32S3","ty":0,"if":0,"ofln":"Offline","onln":"nline","state":["OFF","ON","TOGGLE","HOLD"],"sw":"14.2.0","t":"tasmota_DDEDC","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,00,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,nul,null,null,null,null,null,null,null,null,null,null,null,null,null,null,ull],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver:1}

23:55:55.584 RSL: sensors = {"sn":{"Time":"2024-08-18T23:55:55"},"ver":1
```

Send commands with PuTTY:
https://github.com/arendst/Tasmota/issues/2072#issuecomment-457298226

**Reply by nikito7** [reply]
Default safeboot misses EthType command

Possible fix:

```
[env:tasmota32s3-safeboot]
extends                 = env:tasmota32_base
board                   = esp32s3-qio_qspi
board_build.app_partition_name = safeboot
build_flags             = ${env:tasmota32_base.build_flags}
                         -DFIRMWARE_SAFEBOOT
                         -UUSE_ETHERNET
                         -DUSE_ETHERNET
                         -UETH_TYPE
                         -DETH_TYPE=8
lib_extra_dirs          = lib/lib_ssl, lib/libesp32
lib_ignore              = ${safeboot_flags.lib_ignore}

```

#W5500

**Reply by Jason2866** [reply]
The ETH Type is not needed to set at compile time. It is a persistent setting. Just needs to be configured as most other stuff in Tasmota. It is not working since currently Ethernet support is only enabled in esp32

**Reply by nikito7** [reply]
A working tweak:
  -DUSE_SPI
  -DUSE_ETHERNET

```
[env:tasmota32s3eth-safeboot]
extends                 = env:tasmota32_base
board                   = esp32s3-qio_qspi
board_build.app_partition_name = safeboot
build_flags             = ${env:tasmota32_base.build_flags}
                         -DFIRMWARE_SAFEBOOT
                         -DHAN_S3_ETH
                         -DUSE_SPI
                         -DUSE_ETHERNET
                         -UETH_TYPE
                         -DETH_TYPE=8
lib_extra_dirs          = lib/lib_ssl, lib/libesp32
lib_ignore              = ${safeboot_flags.lib_ignore}

```

```
00:00:00.253 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.254 HDW: ESP32-S3 v0.2 
00:00:00.264 CFG: Loaded from NVS, Count 19
00:00:00.267 SRC: Restart
00:00:00.268 Project tasmota - Tasmota Version 14.2.0.3(safeboot)-3_0_4(2024-09-26T22:28:37)
00:00:00.268 WARNING This version does not support persistent settings
00:00:00.306 ETH: Attempting connection...
00:00:02.043 ETH: Connected at 100Mbps Full Duplex, Mac **** , Hostname Easy-HAN-EB99-eth
00:00:02.055 ETH: IPv4 10.1.0.54, mask 255.255.255.0, gateway 10.1.0.1
00:00:02.444 HTP: Web server active on Easy-HAN-EB99-eth with IP address 10.1.0.54
00:00:02.691 NTP: Sync time...
00:00:02.698 WIF: DNS resolved '2.pool.ntp.org' (85.138.124.211) in 5 ms
00:00:02.734 RTC: UTC 2024-09-26T23:07:22Z, DST 2024-03-31T01:00:00, STD 2024-10-27T02:00:00
00:07:22.000 RTC: Synced by NTP
```

Reset defaults

```
00:00:00.253 CMD: Fall back to serial port, no SOF packet detected on USB port
00:00:00.254 HDW: ESP32-S3 v0.2 
00:00:00.264 CFG: Loaded from NVS, Count 4
00:00:00.267 SRC: Restart
00:00:00.268 Project tasmota - Tasmota Version 14.2.0.3(safeboot)-3_0_4(2024-09-26T22:28:37)
00:00:00.269 WARNING This version does not support persistent settings
00:00:00.316 ETH: Attempting connection...
00:00:00.465 WIF: Checking connection...
00:00:00.565 WIF: WifiManager active for 3 minutes
00:00:00.719 WIF: Wifimanager set AccessPoint
00:00:00.222 HTP: Blank Device - Initial Configuration
00:00:00.224 HTP: Web server active on Easy-HAN-EB99 with IP address 192.168.4.1
00:00:02.053 ETH: Connected at 100Mbps Full Duplex, Mac ****, Hostname Easy-HAN-EB99-eth
00:00:02.065 ETH: IPv4 10.1.0.54, mask 255.255.255.0, gateway 10.1.0.1
00:00:02.401 HTP: Web server active on Easy-HAN-EB99-eth with IP address 10.1.0.54
00:00:02.703 NTP: Sync time...
00:00:02.731 WIF: DNS resolved '2.pool.ntp.org' (85.138.124.211) in 26 ms
00:00:02.768 RTC: UTC 2024-09-26T23:18:55Z, DST 2024-03-31T01:00:00, STD 2024-10-27T02:00:00
00:18:55.001 RTC: Synced by NTP
```

**Comment by LucidEye** [comment]
Forgive me for cutting in on this old thread, but I am curious if Tasmota on an ESP32 can be configured to use an Ethernet connection with a W5550 to connect to a network switch on my LAN, and then have the esp's WiFi work as an AP for the LAN. Basically like an Ethernet-wired range extender AP for the main network.  Thanks in advance for any advice.
![IMG_20260204_005434](https://github.com/user-attachments/assets/64d251c7-c0df-4483-8d38-c5812ef067b4)

**Reply by sfromis** [reply]
Tasmota does not support acting as an AP, outside of initial configuration setup for providing Wifi details. Or via the Range Extender feature. Neither means getting a WiFi/Ethernet bridge.

In regular STA mode, Tasmota will listen and repspond on both interfaces (including different host names), but you can't select interface when opening a connection. And no feature for bridging via STA mode.

There exists a [hack](https://github.com/arendst/Tasmota/discussions/13362) for trying to use Tasmota in AP instead of STA mode, not sure if/how that works if also having Ethernet, but assuming that you even *can* use Ethernet at the same time, that's still no bridge feature built-in. And that hack is not part of the Tasmota project.
